src/main/java/seedu/utility/CurrencyManager.java

```diff
@@ -0,0 +1,85 @@
+package seedu.utility;
+
+import seedu.budget.Budget;
+import seedu.commands.currency.CurrencyType;
+import seedu.entry.Entry;
+import seedu.exceptions.SameCurrencyTypeException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class CurrencyManager {
+
+    protected double exchangeRate;
+    private CurrencyType currency = CurrencyType.SGD;
+    private ArrayList<CurrencyType> currencyTypes = new ArrayList<>();
+
+    public void currencyConvertor(CurrencyType from, CurrencyType to, FinancialTracker finances,
+                                   BudgetManager budgetManager) throws SameCurrencyTypeException {
+        if (getCurrency() == to) {
+            throw new SameCurrencyTypeException(Messages.SAME_CURRENCY_TYPE_MESSAGE);
+        } else {
+            ArrayList<Entry> entries = finances.getEntries();
+            ArrayList<Budget> budgets = budgetManager.getBudgets();
+            convertEntries(entries, from, to);
+            convertBudgets(budgets, from, to);
+        }
+    }
+
+    public CurrencyType getCurrency() {
+        assert currency != null;
+        return currency;
+    }
+
+    public void setCurrency(CurrencyType currency) {
+        this.currency = currency;
+    }
+
+    public ArrayList<CurrencyType> getCurrencyTypes() {
+        currencyTypes.addAll(Arrays.asList(CurrencyType.values()));
+        return currencyTypes;
+    }
+
+    public double determineExchangeRate(CurrencyType to) {
+        switch (to) {
+        case USD:
+            return exchangeRate = 0.74;
+        case SGD:
+            return exchangeRate = 1.35;
+        default:
+            return exchangeRate = 1.00;
+        }
+    }
+
+    public boolean isBaseCurrency(CurrencyType from) {
+        return !from.equals(CurrencyType.SGD);
+    }
+
+    public void convertEntries(ArrayList<Entry> entries, CurrencyType from, CurrencyType to) {
+        for (Entry entry : entries) {
+            double newValue = convertItem(from, to, entry.getValue());
+            assert newValue >= 0;
+            entry.setValue(newValue);
+        }
+        setCurrency(to);
+    }
+
+    public void convertBudgets(ArrayList<Budget> budgets, CurrencyType from, CurrencyType to) {
+        for (Budget budget : budgets) {
+            double newValue = convertItem(from, to, budget.getLimit());
+            budget.setLimit(newValue);
+        }
+        setCurrency(to);
+    }
+
+    public double convertItem(CurrencyType from, CurrencyType to, double value) {
```

Comment:
The conversion process seems alittle odd to me. Not sure if I'm understanding it correctly.